transport/TransportImpl.cpp

```diff
@@ -1076,24 +1076,12 @@ void TransportImpl::internalIceReceived(Endpoint& endpoint,
     auto dataReceiver = _dataReceiver.load();
     if (_rtpIceSession->isAttached(&endpoint))
     {
-        if (ice::isResponse(packet->get()) && _rtpIceSession->isResponseAuthentic(packet->get(), packet->getLength()))
-        {
-        }
-        else if (ice::isRequest(packet->get()) &&
-            _rtpIceSession->isRequestAuthentic(packet->get(), packet->getLength()))
-        {
-        }
-        else
+        if (!_rtpIceSession->isIceAuthentic(packet->get(), packet->getLength()))
         {
             // TODO attack, add metric for this
             return;
         }
```

Comment:
Just trying to understand the situation the change is for. 
Before: new candidates were not accpted after reflexiveProbeTimeout (15s) had passed, unless the nominated probe did not receive responses for 5s, we did accept new candidates regardless of how long into the session we were.
What is the new situation this fix is for? is it to maintain multiple alternative candidates or otherwise the remote side will discard them too, and there is not fail over when nominated fails?

Reply:
Is the description I added helped you to understand the change? The problem was mainly caused by removing PRFLX candidates and only accept them when we were in CONNECTING or until to reflexiveProbeTimeout. The problem is SMB usually takes longer to realise the nominated is not reachable than WebRTC client. And WebRTC client will probe immediately all candidates when it detects the dominated is not reachable. At that moment, SMB can be on CONNECTED state still and not re-add the candidates. When it switch to CONNECTING, it will not have PRFLX backup candidates.

But, in first place, the problem was caused by removing the PRFLX candidates after nomination. Continue accepting is just an improvement for support GATHER_CONTINUALLY where clients can gather candidates anytime and we want to add them as backup.